restli-server/src/main/java/com/linkedin/restli/server/validation/RestLiValidationFilter.java

```diff
@@ -449,8 +455,22 @@ private void validateCreateCollectionResponse(RestLiDataValidator validator,
     }
   }
 
-  private boolean shouldValidateOnResponse(FilterRequestContext requestContext)
+  protected boolean shouldValidateOnRequest(FilterRequestContext requestContext)
+  {
+    // Skip request validation if the header to skip request validation is set.
+    return !(Boolean.TRUE.toString().equals(
+        requestContext.getRequestHeaders().get(RestConstants.HEADER_SKIP_REQUEST_VALIDATION)));
```

Comment:
This is dangerous. A client can simply add this header and skip server side validation. We should not roll this out to all services.

We should add a hook to this filter that a single service can customize and use a header specific to that service to skip validation.

Reply:
Arun mentioned we have a similar case in deco and we worked with ATS to block that header so it cannot be set from requests from outside of our network. We probably should do the similar thing here.

Reply:
Discussed offline.

Raised https://github.com/linkedin/rest.li/pull/676 to address this issue.